### DISCUSSION TITLE
Issue with GPU selection in benchmark execution

### DISCUSSION THREAD

**Discussion by Kranti-Ingale** [sarchlab/mgpusim#125] [Q&A] [answered]
I ran the following command to specify GPU3:
./fir -report-all -verify -timing -trace-mem gpus=3

However, when I checked the mem.trace file, it shows that the execution happened on GPU1 instead of GPU3.

Is there something I'm missing in the configuration, or could this be a bug?

**Comment by DX990307** [comment]
while I think you do not specify the benchmark size, could you try the following instruction: ./fir -report-all -verify -timing -trace-mem gpus=3 -length=2048

**Comment by syifan** [comment, marked as answer]
It looks like the problem is that you forgot a dash before `gpus`. So you need this command 

```
./fir -report-all -verify -timing -trace-mem -gpus=3
```

**Reply by Kranti-Ingale** [reply]
Thank you.

**Reply by Kranti-Ingale** [reply]
Does running  ./kmeans -report-all -verify -timing -trace-mem -gpus=2,3 mean that only GPUs 2 and 3 are used while GPUs 1 and 4 remain idle? I noticed that in some cases, the `mem.trace` file shows GPU 1 being active, but not consistently. Could you please clarify why this happens?

**Reply by syifan** [reply]
Only GPU 2 and 3 will be used. GPU 4 will not even be created. 

I also see a few GPU[1] tasks. However, I do not know why there are only task ends but no start. This looks like a bug to me. Very likely, these are only cache flushes at kernel boundaries and should not have any real overall timing impacts.
